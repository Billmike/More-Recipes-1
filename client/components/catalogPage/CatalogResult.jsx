import React from 'react';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroll-component';

import RecipeCatalogCard from '../common/RecipeCatalogCard';
import Spinner from '../common/Spinner';

const propTypes = {
  recipes: PropTypes.shape({
    recipes: PropTypes.arrayOf(PropTypes.shape()),
    hasMore: PropTypes.bool.isRequired,
  }).isRequired,
  retrieveMoreRecipes: PropTypes.func.isRequired,
};

/**
 * Function representing catalog result component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {object} React element
 */
const CatalogResult = (props) => {
  const { recipes: { recipes, hasMore }, retrieveMoreRecipes } = props;

  return (
    <div className="row">
      <div className="catalog-page">
        <InfiniteScroll
          next={retrieveMoreRecipes}
          hasMore={hasMore}
          loader={<div className="col s12 center-align">
            <Spinner size="small" />
          </div>
          }
          endMessage={
            <div className="end-message center-align col s12">
              <b>Yay! You have seen it all</b>
            </div>
          }
        >
          {recipes && recipes.map((recipe, index) =>
            (<RecipeCatalogCard
              {...props}
              key={recipe.id}
              index={index}
              recipe={recipe}
            />))
          }
        </InfiniteScroll>
      </div>
    </div>
  );
};

CatalogResult.propTypes = propTypes;

export default CatalogResult;
