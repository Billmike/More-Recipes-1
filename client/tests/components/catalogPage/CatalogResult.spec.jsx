import React from 'react';

import CatalogResult from '../../../components/catalogPage/CatalogResult';

describe('CatalogResult component', () => {
  it('should render component', () => {
    const props = {
      recipes: {
        recipes: [{
          title: 'Beans',
          category: 'Main',
          description: 'Tasty beans',
          preparationTime: 85,
          ingredients: 'Beans, onions, palm oil, salt',
          directions: 'Do this. Do that',
        }],
        hasMore: false,
      },
      retrieveMoreRecipes: jest.fn(),
    };
    const wrapper = shallow(<CatalogResult {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
