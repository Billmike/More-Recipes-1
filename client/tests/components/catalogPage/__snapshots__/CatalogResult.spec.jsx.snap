// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogResult component should render component 1`] = `
<div
  className="row"
>
  <div
    className="catalog-page"
  >
    <InfiniteScroll
      disableBrowserPullToRefresh={true}
      endMessage={
        <div
          className="end-message center-align col s12"
        >
          <b>
            Yay! You have seen it all
          </b>
        </div>
      }
      hasMore={false}
      loader={
        <div
          className="col s12 center-align"
        >
          <Spinner
            size="small"
          />
        </div>
      }
      next={[MockFunction]}
      pullDownToRefreshContent={
        <h3>
          Pull down to refresh
        </h3>
      }
      pullDownToRefreshThreshold={100}
      releaseToRefreshContent={
        <h3>
          Release to refresh
        </h3>
      }
    >
      <RecipeCatalogCard
        index={0}
        recipe={
          Object {
            "category": "Main",
            "description": "Tasty beans",
            "directions": "Do this. Do that",
            "ingredients": "Beans, onions, palm oil, salt",
            "preparationTime": 85,
            "title": "Beans",
          }
        }
        recipes={
          Object {
            "hasMore": false,
            "recipes": Array [
              Object {
                "category": "Main",
                "description": "Tasty beans",
                "directions": "Do this. Do that",
                "ingredients": "Beans, onions, palm oil, salt",
                "preparationTime": 85,
                "title": "Beans",
              },
            ],
          }
        }
        retrieveMoreRecipes={[MockFunction]}
      />
    </InfiniteScroll>
  </div>
</div>
`;
